test/src/unit-capi-metadata.cc

```diff
@@ -332,6 +332,69 @@ TEST_CASE_METHOD(
   tiledb_array_free(&array);
 }
 
+TEST_CASE_METHOD(
+    CMetadataFx,
+    "C API: Metadata, concurrent writes",
+    "[!mayfail][capi][metadata][concurrent_writes]") {
+  int32_t one = 1;
+  int32_t two = 2;
+  int fail_count = 0;
+  const void* v_r = nullptr;
+  tiledb_datatype_t v_type;
+  uint32_t v_num;
+
+  // Run the test body 1000 times
```

Comment:
How long does the test take to run? I'd like to find the right value here.

Reply:
Not long but 1k runs is certainly excessive. Failure rate (at least locally) always seems to hover around 25%. 
Does 20 sound like a reasonable number?

Reply:
Updated to 20 for now. Pending review / approval

Reply:
I looked in CI. Some runners only failed one iteration. 100 times is probably better as it won't take longer than a second to run from what I can see.